src/odemis/acq/stream/_base.py

```diff
@@ -990,6 +1001,13 @@ class NeedRecomputeException(Exception):
                             # but using the cache from the last execution
                             raise NeedRecomputeException()
 
+                        # the projected tiles cache is invalid
```

Comment:
Don't have two checks for when to recompute. Instead:

- Move the `if self._projectedTilesInvalid` (+ reset cache) to the beginning of the while loop.
- Check whether to recompute just based on `if _im_needs_recompute.is_set()`.

One fix that needs to be done for this to work properly: move `self._shouldUpdateImage()` from the setters of mpp and .rect to (new) subscribers of .mpp and .rect (which ensures that the new .value is set).